pkg/services/sqlstore/dashboard_librarypanels.go

```diff
@@ -0,0 +1,35 @@
+package sqlstore
+
+import (
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+func init() {
+	bus.AddHandler("sql", GetLibraryPanelsForDashboardID)
+}
+
+func GetLibraryPanelsForDashboardID(query *models.GetLibraryPanelsQuery) error {
+	sql := `SELECT
+				lp.id, lp.org_id, lp.folder_id, lp.uid, lp.name, lp.model, lp.created, lp.created_by, lp.updated, updated_by
+			FROM
+				library_panel_dashboard lpd
+			INNER JOIN
+				library_panel lp ON lpd.librarypanel_id = lp.id AND lpd.dashboard_id=?`
+
+	libraryPanels := make([]models.LibraryPanel, 0)
+	sess := x.SQL(sql, query.DashboardId)
+	err := sess.Find(&libraryPanels)
+	if err != nil {
+		return err
+	}
+
+	libraryPanelMap := make(map[string]models.LibraryPanel)
+	for _, panel := range libraryPanels {
+		libraryPanelMap[panel.UID] = panel
+	}
+
+	query.Result = libraryPanelMap
+
+	return nil
+}
```

Comment:
Maybe this should be in the librarypanel service instead? I don't know what would be the preferred approach here because the library panel service doesn't use the bus at all. Should librarypanel service be a dependency to dashboard? I feel like my backend skills are a bit weak to reason about this and could need help here.

Reply:
think you can move this to a PanelLibraryService  function

Reply:
Yepp will do that. Thank you for clarifying!